clients/src/main/java/org/apache/kafka/common/requests/FetchResponse.java

```diff
@@ -240,9 +269,26 @@ public String toString() {
         public final long highWatermark;
         public final long lastStableOffset;
         public final long logStartOffset;
+        public final Optional<Integer> preferredReadReplica;
         public final List<AbortedTransaction> abortedTransactions;
         public final T records;
 
+        public PartitionData(Errors error,
+                             long highWatermark,
+                             long lastStableOffset,
+                             long logStartOffset,
+                             Integer preferredReadReplica,
```

Comment:
Maybe we should make this type `Optional<Integer>`. Otherwise it's a little confusing if you should use `null` or `UNSPECIFIED_PREFERRED_REPLICA`.